werkzeug/routing.py

```diff
@@ -1607,6 +1630,23 @@ def _partial_build(self, endpoint, values, method, append_unknown):
                 if rv is not None:
                     return rv
 
+    def _closest_rule(self, endpoint, values, method):
```

Comment:
I think this helper should be moved to the exception class, because there's already too many methods here.

If you moved this helper here because you want to override it for your own usecases, I rather suggest adding the ability to specify a custom BuildError subclass.